src/services/goToDefinition.ts

```diff
@@ -595,7 +597,8 @@ function isExpandoDeclaration(node: Declaration): boolean {
 }
 
 function getDefinitionFromSymbol(typeChecker: TypeChecker, symbol: Symbol, node: Node, failedAliasResolution?: boolean, declarationFilter?: (d: Declaration) => boolean): DefinitionInfo[] | undefined {
-    const filteredDeclarations = declarationFilter !== undefined ? filter(symbol.declarations, declarationFilter) : symbol.declarations;
+    const declarations = isTransientSymbol(symbol) && symbol.links.checkFlags & CheckFlags.Mapped && !symbol.declarations?.length ? symbol.links.syntheticOrigin?.declarations : symbol.declarations;

```

Comment:
I find it sort of concerning to reach into the `links` like this; I can see maybe one or two other cases where we've done `symbol.links` outside the checker and I can't say I like those either. Is there no other way to do this?

Reply:
I agree with your concern about reaching into symbol.links. I think there might be a better way to handle this. Maybe we can create a helper function that gets the documentation from the original symbol without directly accessing links. That would make the code more maintainable and less dependent on internal implementation details.